websites/A/Animes Portal/presence.ts

```diff
@@ -0,0 +1,294 @@
+const presence = new Presence({
+  clientId: "924791712944099389"
+});
+
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+      largeImageKey: "logo"
+    },
+    { pathname } = document.location,
+    paths = pathname.split("/");
+  if (!paths[0]) paths.shift();
+
+  if (pathname === "/") presenceData.details = "Viewing the home page";
+  else if (pathname.startsWith("/messages")) {
+    if (!paths[1]) presenceData.details = "Viewing messages";
+    else if (paths[1].startsWith("pm-")) {
+      const body = document.querySelector(
+          "body > main.animated > div.wrapper > div.dialogPadding"
+        ),
+        username = body.querySelector("span.username").textContent;
+
+      presenceData.smallImageKey = parseAvatarFromAttr(
+        body.querySelector("a.avatar").attributes.getNamedItem("style")
+          .textContent
+      );
+
+      presenceData.details = `Messaging ${username}`;
+      presenceData.smallImageText = username;
```

Comment:
No need to display the username twice as it is displayed in presenceData.details on the line above
```suggestion
```

Reply:
It's looking better when we have username on `smallImageText`

Reply:
It is repetitive